src/v/coproc/event_listener.cc

```diff
@@ -252,4 +175,35 @@ event_listener::poll_topic(model::record_batch_reader::data_t& events) {
                                  : ss::stop_iteration::no;
 };
 
+ss::future<>
+event_listener::process_events(event_batch events, model::offset last_offset) {
+    for (auto& [type, prepared_events] : events) {
+        auto it = _handlers.find(type);
+        if (it == _handlers.end()) {
+            vlog(
+              coproclog.warn,
+              "Can not find handler for coproc_type: {}",
+              coproc_type_as_string_view(type));
+            continue;
+        }
+
+        switch (type) {
```

Comment:
This can be refactored for clarity:
```
try {
  co_await it->second->process(std::move(prepared_events));
} catch(const event_handler_exception &ex) {
  switch(type){
  case coproc_type::nodejs:
    _offset = last_offset;
  default: vlog(...);
  }
}
```

Reply:
done